commit c3a516b42fbebf19bbfb33e1e55c04b7831fc4ac
Author: Simon Kaegi <skaegi>
Date:   Wed Nov 26 23:46:39 2008 +0000

    Bug 256650 test failures in N20081126-2000 - Error while loading manipulator

29	5	bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/ParserUtils.java
8	0	bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/utils/FileUtils.java
2	1	bundles/org.eclipse.equinox.p2.touchpoint.eclipse/src/org/eclipse/equinox/internal/p2/touchpoint/eclipse/Util.java
diff --git a/bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/ParserUtils.java b/bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/ParserUtils.java
index 91b85c7..93efb3a 100644
--- a/bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/ParserUtils.java
+++ b/bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/ParserUtils.java
@@ -20,6 +20,8 @@ import org.eclipse.equinox.internal.provisional.frameworkadmin.LauncherData;
 import org.osgi.service.log.LogService;
 
 public class ParserUtils {
+	private static final String FILE_PROTOCOL = "file:"; //$NON-NLS-1$
+
 	public static File getOSGiInstallArea(List programArgs, Properties properties, LauncherData launcherData) {
 		if (launcherData == null)
 			return null;
@@ -58,13 +60,35 @@ public class ParserUtils {
 		if (args == null)
 			return null;
 		String install = getValueForArgument(EquinoxConstants.OPTION_INSTALL, args);
-		if (install != null)
-			return new File(install);
-		if (properties != null && properties.getProperty("osgi.install.area") != null)
-			return new File(properties.getProperty("osgi.install.area"));
+		if (install == null && properties != null)
+			install = properties.getProperty("osgi.install.area");
+
+		if (install != null) {
+			if (install.startsWith(FILE_PROTOCOL))
+				install = install.substring(FILE_PROTOCOL.length() + 1);
+			File installFile = new File(install);
+			if (installFile.isAbsolute())
+				return installFile;
+			return URIUtil.toFile(URIUtil.makeAbsolute(installFile.toURI(), base));
+		}
+
 		String startup = getValueForArgument(EquinoxConstants.OPTION_STARTUP, args);
 		if (startup != null && base != null) {
-			return URIUtil.toFile(URIUtil.makeAbsolute(fromOSGiJarToOSGiInstallArea(startup).toURI(), base));
+			if (startup.startsWith(FILE_PROTOCOL)) {
+				try {
+					URI startupURI = new URI(startup);
+					startup = new File(startupURI).getAbsolutePath();
+				} catch (URISyntaxException e) {
+					startup = startup.substring(FILE_PROTOCOL.length() + 1);
+				}
+			}
+
+			File osgiInstallArea = fromOSGiJarToOSGiInstallArea(startup);
+			if (osgiInstallArea.isAbsolute())
+				return osgiInstallArea;
+
+			File baseFile = new File(base);
+			return new File(baseFile, osgiInstallArea.getPath());
 		}
 		return null;
 	}
diff --git a/bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/utils/FileUtils.java b/bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/utils/FileUtils.java
index 69662f9..64875fe 100644
--- a/bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/utils/FileUtils.java
+++ b/bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/utils/FileUtils.java
@@ -158,6 +158,14 @@ public class FileUtils {
 	}
 
 	public static URI fromPath(String path) throws URISyntaxException {
+		if (path.startsWith(FILE_PROTOCOL)) {
+			try {
+				return new URI(path);
+			} catch (URISyntaxException e) {
+				path = path.substring(FILE_PROTOCOL.length() + 1);
+			}
+		}
+
 		File f = new File(path);
 		if (f.isAbsolute())
 			return f.toURI();
diff --git a/bundles/org.eclipse.equinox.p2.touchpoint.eclipse/src/org/eclipse/equinox/internal/p2/touchpoint/eclipse/Util.java b/bundles/org.eclipse.equinox.p2.touchpoint.eclipse/src/org/eclipse/equinox/internal/p2/touchpoint/eclipse/Util.java
index fdcfde4..c7ed17e 100644
--- a/bundles/org.eclipse.equinox.p2.touchpoint.eclipse/src/org/eclipse/equinox/internal/p2/touchpoint/eclipse/Util.java
+++ b/bundles/org.eclipse.equinox.p2.touchpoint.eclipse/src/org/eclipse/equinox/internal/p2/touchpoint/eclipse/Util.java
@@ -162,7 +162,8 @@ public class Util {
 
 	public static BundleInfo createBundleInfo(File bundleFile, String manifest) {
 		BundleInfo bundleInfo = new BundleInfo();
-		bundleInfo.setLocation(bundleFile.toURI());
+		if (bundleFile != null)
+			bundleInfo.setLocation(bundleFile.toURI());
 
 		bundleInfo.setManifest(manifest);
 		try {
commit 820f40be999fc62feaa39a67515e2ea7aeb45281
Author: Simon Kaegi <skaegi>
Date:   Thu Nov 27 00:41:38 2008 +0000

    Bug 256650 test failures in N20081126-2000 - Error while loading manipulator

8	8	bundles/org.eclipse.equinox.p2.touchpoint.eclipse/src/org/eclipse/equinox/internal/p2/touchpoint/eclipse/EclipseTouchpoint.java
diff --git a/bundles/org.eclipse.equinox.p2.touchpoint.eclipse/src/org/eclipse/equinox/internal/p2/touchpoint/eclipse/EclipseTouchpoint.java b/bundles/org.eclipse.equinox.p2.touchpoint.eclipse/src/org/eclipse/equinox/internal/p2/touchpoint/eclipse/EclipseTouchpoint.java
index dc6fbf0..54104ba 100644
--- a/bundles/org.eclipse.equinox.p2.touchpoint.eclipse/src/org/eclipse/equinox/internal/p2/touchpoint/eclipse/EclipseTouchpoint.java
+++ b/bundles/org.eclipse.equinox.p2.touchpoint.eclipse/src/org/eclipse/equinox/internal/p2/touchpoint/eclipse/EclipseTouchpoint.java
@@ -101,30 +101,30 @@ public class EclipseTouchpoint extends Touchpoint {
 	}
 
 	public IStatus commit(IProfile profile) {
+		MultiStatus status = new MultiStatus(Activator.ID, IStatus.OK, null, null);
 		try {
 			saveManipulator(profile);
 		} catch (RuntimeException e) {
-			return Util.createError(Messages.error_saving_manipulator, e);
+			status.add(Util.createError(Messages.error_saving_manipulator, e));
 		} catch (IOException e) {
-			return Util.createError(Messages.error_saving_manipulator, e);
+			status.add(Util.createError(Messages.error_saving_manipulator, e));
 		}
 		try {
 			savePlatformConfigurationWrapper(profile);
 		} catch (RuntimeException e) {
-			return Util.createError(Messages.error_saving_platform_configuration, e);
+			status.add(Util.createError(Messages.error_saving_platform_configuration, e));
 		} catch (ProvisionException pe) {
-			return Util.createError(Messages.error_saving_platform_configuration, pe);
+			status.add(Util.createError(Messages.error_saving_platform_configuration, pe));
 		}
 
 		try {
 			saveSourceManipulator(profile);
 		} catch (RuntimeException e) {
-			return Util.createError(Messages.error_saving_source_bundles_list, e);
+			status.add(Util.createError(Messages.error_saving_source_bundles_list, e));
 		} catch (IOException e) {
-			return Util.createError(Messages.error_saving_source_bundles_list, e);
+			status.add(Util.createError(Messages.error_saving_source_bundles_list, e));
 		}
-
-		return Status.OK_STATUS;
+		return status;
 	}
 
 	public IStatus rollback(IProfile profile) {
commit af87fe18f1bc198d212c104bdb35dfc0ce846634
Author: Simon Kaegi <skaegi>
Date:   Thu Nov 27 01:23:47 2008 +0000

    Bug 256650 test failures in N20081126-2000 - Error while loading manipulator
    
    Support old style bundle locations  when a simpleconfigurator with version less than 1.0.100 is present so that end 2 end tests pass

3	2	bundles/org.eclipse.equinox.frameworkadmin.test/src/org/eclipse/equinox/frameworkadmin/tests/RelativePathTest.java
4	4	bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtilsTest.java
24	4	bundles/org.eclipse.equinox.simpleconfigurator.manipulator/src/org/eclipse/equinox/internal/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtils.java
diff --git a/bundles/org.eclipse.equinox.frameworkadmin.test/src/org/eclipse/equinox/frameworkadmin/tests/RelativePathTest.java b/bundles/org.eclipse.equinox.frameworkadmin.test/src/org/eclipse/equinox/frameworkadmin/tests/RelativePathTest.java
index 38d26b6..d83f784 100644
--- a/bundles/org.eclipse.equinox.frameworkadmin.test/src/org/eclipse/equinox/frameworkadmin/tests/RelativePathTest.java
+++ b/bundles/org.eclipse.equinox.frameworkadmin.test/src/org/eclipse/equinox/frameworkadmin/tests/RelativePathTest.java
@@ -63,8 +63,9 @@ public class RelativePathTest extends FwkAdminAndSimpleConfiguratorTest {
 		assertNotContent(new File(configurationFolder, "config.ini"), installFolder.getAbsolutePath());
 		assertNotContent(new File(configurationFolder, "org.eclipse.equinox.simpleconfigurator/bundles.info"), installFolder.getAbsolutePath());
 		assertContent(new File(configurationFolder, "config.ini"), ":org.eclipse.equinox.simpleconfigurator.jar");
-		assertContent(new File(configurationFolder, "org.eclipse.equinox.simpleconfigurator/bundles.info"), ",plugins/org.eclipse.equinox.simpleconfigurator.jar");
-		assertContent(new File(configurationFolder, "org.eclipse.equinox.simpleconfigurator/bundles.info"), ",plugins/org.eclipse.osgi.jar");
+		//Note: This is testing for old style bundle locations
+		assertContent(new File(configurationFolder, "org.eclipse.equinox.simpleconfigurator/bundles.info"), "file:plugins/org.eclipse.equinox.simpleconfigurator.jar");
+		assertContent(new File(configurationFolder, "org.eclipse.equinox.simpleconfigurator/bundles.info"), "file:plugins/org.eclipse.osgi.jar");
 
 		BundleInfo bi = new BundleInfo(new URI(FileLocator.resolve(Activator.getContext().getBundle().getEntry("dataFile/bundle_1")).toExternalForm()), 2);
 		manipulator.getConfigData().addBundle(bi);
diff --git a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtilsTest.java b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtilsTest.java
index e3a2d51..b185286 100644
--- a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtilsTest.java
+++ b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtilsTest.java
@@ -23,7 +23,7 @@ public class SimpleConfiguratorManipulatorUtilsTest extends AbstractProvisioning
 	public void testWriteBundleInfoLine() throws URISyntaxException, IOException {
 		String expectedLine = "javax.servlet,2.4.0.v200806031604,plugins/javax.servlet_2.4.0.v200806031604.jar,4,false";
 		BundleInfo bundleInfo = new BundleInfo("javax.servlet", "2.4.0.v200806031604", new URI("plugins/javax.servlet_2.4.0.v200806031604.jar"), 4, false);
-		String line = SimpleConfiguratorManipulatorUtils.createBundleInfoLine(bundleInfo);
+		String line = SimpleConfiguratorManipulatorUtils.createBundleInfoLine(bundleInfo, false);
 		assertEquals(expectedLine, line);
 		assertEquals(bundleInfo, SimpleConfiguratorUtils.parseBundleInfoLine(line, null));
 	}
@@ -31,7 +31,7 @@ public class SimpleConfiguratorManipulatorUtilsTest extends AbstractProvisioning
 	public void testWriteBundleInfoLineWithComma() throws URISyntaxException, IOException {
 		String expectedLine = "javax.servlet,2.4.0.v200806031604,plugin%2Cs/javax.servlet_2.4.0.v200806031604.jar,4,false";
 		BundleInfo bundleInfo = new BundleInfo("javax.servlet", "2.4.0.v200806031604", new URI("plugin,s/javax.servlet_2.4.0.v200806031604.jar"), 4, false);
-		String line = SimpleConfiguratorManipulatorUtils.createBundleInfoLine(bundleInfo);
+		String line = SimpleConfiguratorManipulatorUtils.createBundleInfoLine(bundleInfo, false);
 		assertEquals(expectedLine, line);
 		assertEquals(bundleInfo, SimpleConfiguratorUtils.parseBundleInfoLine(line, null));
 	}
@@ -39,7 +39,7 @@ public class SimpleConfiguratorManipulatorUtilsTest extends AbstractProvisioning
 	public void testWriteBundleInfoLineWithSpace() throws URISyntaxException, IOException {
 		String expectedLine = "javax.servlet,2.4.0.v200806031604,plugin%20s/javax.servlet_2.4.0.v200806031604.jar,4,false";
 		BundleInfo bundleInfo = new BundleInfo("javax.servlet", "2.4.0.v200806031604", new URI("plugin%20s/javax.servlet_2.4.0.v200806031604.jar"), 4, false);
-		String line = SimpleConfiguratorManipulatorUtils.createBundleInfoLine(bundleInfo);
+		String line = SimpleConfiguratorManipulatorUtils.createBundleInfoLine(bundleInfo, false);
 		assertEquals(expectedLine, line);
 		assertEquals(bundleInfo, SimpleConfiguratorUtils.parseBundleInfoLine(line, null));
 	}
@@ -47,7 +47,7 @@ public class SimpleConfiguratorManipulatorUtilsTest extends AbstractProvisioning
 	public void testWriteBundleInfoLineWithAtSymbol() throws URISyntaxException, IOException {
 		String expectedLine = "javax.servlet,2.4.0.v200806031604,plugin@s/javax.servlet_2.4.0.v200806031604.jar,4,false";
 		BundleInfo bundleInfo = new BundleInfo("javax.servlet", "2.4.0.v200806031604", new URI("plugin@s/javax.servlet_2.4.0.v200806031604.jar"), 4, false);
-		String line = SimpleConfiguratorManipulatorUtils.createBundleInfoLine(bundleInfo);
+		String line = SimpleConfiguratorManipulatorUtils.createBundleInfoLine(bundleInfo, false);
 		assertEquals(expectedLine, line);
 		assertEquals(bundleInfo, SimpleConfiguratorUtils.parseBundleInfoLine(line, null));
 	}
diff --git a/bundles/org.eclipse.equinox.simpleconfigurator.manipulator/src/org/eclipse/equinox/internal/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtils.java b/bundles/org.eclipse.equinox.simpleconfigurator.manipulator/src/org/eclipse/equinox/internal/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtils.java
index 183a5aa..d01a331 100644
--- a/bundles/org.eclipse.equinox.simpleconfigurator.manipulator/src/org/eclipse/equinox/internal/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtils.java
+++ b/bundles/org.eclipse.equinox.simpleconfigurator.manipulator/src/org/eclipse/equinox/internal/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtils.java
@@ -14,9 +14,12 @@ import java.util.Arrays;
 import java.util.Comparator;
 import org.eclipse.equinox.internal.frameworkadmin.utils.Utils;
 import org.eclipse.equinox.internal.simpleconfigurator.utils.BundleInfo;
+import org.osgi.framework.Version;
 
 public class SimpleConfiguratorManipulatorUtils {
 
+	private static final Version OLD_VERSION = new Version("1.0.100");
+
 	public static void writeConfiguration(BundleInfo[] simpleInfos, File outputFile) throws IOException {
 
 		// if empty remove the configuration file
@@ -44,10 +47,20 @@ public class SimpleConfiguratorManipulatorUtils {
 		Utils.createParentDir(outputFile);
 		BufferedWriter writer = null;
 		IOException caughtException = null;
+		boolean oldStyle = false;
+		for (int i = 0; i < simpleInfos.length; i++) {
+			if (SimpleConfiguratorManipulatorImpl.SERVICE_PROP_VALUE_CONFIGURATOR_SYMBOLICNAME.equals(simpleInfos[i].getSymbolicName())) {
+				Version version = new Version(simpleInfos[i].getVersion());
+				if (version.compareTo(OLD_VERSION) < 0)
+					oldStyle = true;
+				break;
+			}
+		}
+
 		try {
 			writer = new BufferedWriter(new FileWriter(outputFile));
 			for (int i = 0; i < simpleInfos.length; i++) {
-				writer.write(createBundleInfoLine(simpleInfos[i]));
+				writer.write(createBundleInfoLine(simpleInfos[i], oldStyle));
 				writer.newLine();
 			}
 		} catch (IOException e) {
@@ -67,14 +80,14 @@ public class SimpleConfiguratorManipulatorUtils {
 			throw caughtException;
 	}
 
-	public static String createBundleInfoLine(BundleInfo bundleInfo) throws IOException {
+	public static String createBundleInfoLine(BundleInfo bundleInfo, boolean oldStyle) throws IOException {
 		// symbolicName,version,location,startLevel,markedAsStarted
 		StringBuffer buffer = new StringBuffer();
 		buffer.append(bundleInfo.getSymbolicName());
 		buffer.append(',');
 		buffer.append(bundleInfo.getVersion());
 		buffer.append(',');
-		buffer.append(createBundleLocation(bundleInfo.getLocation()));
+		buffer.append(createBundleLocation(bundleInfo.getLocation(), oldStyle));
 		buffer.append(',');
 		buffer.append(bundleInfo.getStartLevel());
 		buffer.append(',');
@@ -82,7 +95,14 @@ public class SimpleConfiguratorManipulatorUtils {
 		return buffer.toString();
 	}
 
-	public static String createBundleLocation(URI location) {
+	public static String createBundleLocation(URI location, boolean oldStyle) {
+		if (oldStyle) {
+			String scheme = location.getScheme();
+			if (scheme == null)
+				scheme = "file";
+			return scheme + ":" + location.getSchemeSpecificPart();
+		}
+
 		String result = location.toString();
 		int commaIndex = result.indexOf(',');
 		while (commaIndex != -1) {
commit 9572208085ccfed21e1f644aa1458ef3a865a2c3
Author: Simon Kaegi <skaegi>
Date:   Thu Nov 27 02:38:27 2008 +0000

    Bug 256650 test failures in N20081126-2000 - Error while loading manipulator

12	5	bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/EquinoxFwConfigFileParser.java
diff --git a/bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/EquinoxFwConfigFileParser.java b/bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/EquinoxFwConfigFileParser.java
index 6684ba5..8592940 100644
--- a/bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/EquinoxFwConfigFileParser.java
+++ b/bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/EquinoxFwConfigFileParser.java
@@ -173,11 +173,11 @@ public class EquinoxFwConfigFileParser {
 		}
 
 		//Start figuring out stuffs 
-		URI rootURL = launcherData.getLauncher() != null ? launcherData.getLauncher().getParentFile().toURI() : null;
+		URI rootURI = launcherData.getLauncher() != null ? launcherData.getLauncher().getParentFile().toURI() : null;
 
 		readFwJarLocation(configData, launcherData, props);
 		URI configArea = inputFile.getParentFile().toURI();
-		readLauncherPath(props, rootURL);
+		readLauncherPath(props, rootURI);
 		readp2DataArea(props, configArea);
 		readSimpleConfiguratorURL(props, configArea);
 		readBundlesList(manipulator, ParserUtils.getOSGiInstallArea(Arrays.asList(launcherData.getProgramArgs()), props, launcherData).toURI(), props.getProperty(EquinoxConstants.PROP_BUNDLES));
@@ -348,8 +348,13 @@ public class EquinoxFwConfigFileParser {
 
 	private void writeLauncherPath(ConfigData configData, Properties props, URI root) throws URISyntaxException {
 		String value = getFwProperty(configData, EquinoxConstants.PROP_LAUNCHER_PATH);
-		if (value != null)
-			props.setProperty(EquinoxConstants.PROP_LAUNCHER_PATH, URIUtil.toUnencodedString(URIUtil.makeRelative(URIUtil.fromString(value), root)));
+		if (value != null) {
+			URI launcherPathURI = FileUtils.fromPath(value);
+			String launcherPath = URIUtil.toUnencodedString(URIUtil.makeRelative(launcherPathURI, root));
+			if ("/".equals(launcherPath) || "".equals(launcherPath)) //$NON-NLS-1$ //$NON-NLS-2$
+				launcherPath = "."; //$NON-NLS-1$
+			props.setProperty(EquinoxConstants.PROP_LAUNCHER_PATH, launcherPath);
+		}
 	}
 
 	private void readSimpleConfiguratorURL(Properties props, URI configArea) throws URISyntaxException {
@@ -393,9 +398,11 @@ public class EquinoxFwConfigFileParser {
 		String header = "This configuration file was written by: " + this.getClass().getName(); //$NON-NLS-1$
 
 		Properties configProps = new Properties();
+		URI rootURI = launcherData.getLauncher() != null ? launcherData.getLauncher().getParentFile().toURI() : null;
+
 		writeFwJarLocation(configData, launcherData, configProps);
 		try {
-			writeLauncherPath(configData, configProps, null);
+			writeLauncherPath(configData, configProps, rootURI);
 			URI configArea = manipulator.getLauncherData().getFwConfigLocation().toURI();
 			writep2DataArea(configData, configProps, configArea);
 			writeSimpleConfiguratorURL(configData, configProps, configArea);
commit 856132724650fc2010839725366db833b3d53e6f
Author: Simon Kaegi <skaegi>
Date:   Wed Dec 10 00:44:33 2008 +0000

    Bug 256650
     test failures in N20081126-2000 - Error while loading manipulator
    
    I've updated the version of simple configurator for which we write an old style bundle.info to be a SC bundle with version less than 1.0.100.v20081206

2	2	bundles/org.eclipse.equinox.simpleconfigurator.manipulator/src/org/eclipse/equinox/internal/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtils.java
diff --git a/bundles/org.eclipse.equinox.simpleconfigurator.manipulator/src/org/eclipse/equinox/internal/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtils.java b/bundles/org.eclipse.equinox.simpleconfigurator.manipulator/src/org/eclipse/equinox/internal/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtils.java
index 5e3c928..668b7f2 100644
--- a/bundles/org.eclipse.equinox.simpleconfigurator.manipulator/src/org/eclipse/equinox/internal/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtils.java
+++ b/bundles/org.eclipse.equinox.simpleconfigurator.manipulator/src/org/eclipse/equinox/internal/simpleconfigurator/manipulator/SimpleConfiguratorManipulatorUtils.java
@@ -20,7 +20,7 @@ public class SimpleConfiguratorManipulatorUtils {
 
 	private static final String VERSION_PREFIX = "#version=";
 	private static final String VERSION_1 = "1";
-	private static final Version OLD_VERSION = new Version("1.0.100");
+	private static final Version OLD_STYLE_SIMPLE_CONFIGURATOR_VERSION = new Version("1.0.100.v20081206");
 
 	public static void writeConfiguration(BundleInfo[] simpleInfos, File outputFile) throws IOException {
 
@@ -53,7 +53,7 @@ public class SimpleConfiguratorManipulatorUtils {
 		for (int i = 0; i < simpleInfos.length; i++) {
 			if (SimpleConfiguratorManipulatorImpl.SERVICE_PROP_VALUE_CONFIGURATOR_SYMBOLICNAME.equals(simpleInfos[i].getSymbolicName())) {
 				Version version = new Version(simpleInfos[i].getVersion());
-				if (version.compareTo(OLD_VERSION) < 0)
+				if (version.compareTo(OLD_STYLE_SIMPLE_CONFIGURATOR_VERSION) < 0)
 					oldStyle = true;
 				break;
 			}
